README.md

```diff
@@ -1,8 +1,118 @@
+  
+#### Anatomy of configuration  
+(All config file code examples are taken from `/libero-config/config--libero-default.js`.)  
+
+##### Simple example
+`config.data` is where you define your configuration data.
+Here `config.data` defines the the `small` and `medium` site breakpoints:  
+  
+```  
+config.data.breakpoint = {site: {}};  
+config.data.breakpoint.site.small = 480;  
+config.data.breakpoint.site.medium = 730;  
+```  
+   
+`config.layerAllocations` specifies which technology layers the properties of `config.data` are distributed to. Continuing the above example:  
+```  
+config.layerAllocations = {  
+ sass: ['breakpoint'],
+ js: ['breakpoint'],
+ template: ['breakpoint'] };
+ ```  
+specifies that the `breakpoint` config must be distributed to all three available layers: the sass, JavaScript and the templating layer.  
+
+##### Advanced example
+Sometimes configuration values depend on other configuration values, for example measures in a grid system. To be able to maintain these relationships even when the underlying predicate value may be modified by a later-loading config file, the calculation of the final value determined by these relationships must be deferred until all specified configurations are loaded and parsed. This is achieved by specifying these simple mathematical expressions in the format:
+```
+'!expression [some simple mathematical expression]'
+```
+Using this we can specify the baseline grid as:
+```
+config.data.baselinegrid = {space: {}};  
+config.data.baselinegrid.space.extra_small_in_px = 12;  
+config.data.baselinegrid.space.small_in_px = '!expression baselinegrid.space.extra_small_in_px * 2';  
+config.data.baselinegrid.space.smallish_in_px = '!expression baselinegrid.space.small_in_px * 1.5';  
+config.data.baselinegrid.space.medium_in_px = '!expression baselinegrid.space.small_in_px * 2';
+...
+```
+The result is that `config.data.baselinegrid.space.small_in_px` will have the value twice that of whatever the final value of `config.data.baselinegrid.space.extra_small_in_px`is, *even if `config.data.baselinegrid.space.extra_small_in_px` is modified by a later loading config*. This provides a way of reusing the essentials of the baseline grid system, but basing it on a different key value as required.
+
+#### Distributing configuration  
+##### Distributing to SASS  
+Each property of `config.data` specified in `config.layerAllocations.sass` is eventually written as a SASS file to  `/source/css/sass/derived-from-config/_variables--[propertyname].sass`. Each of these files contains the SASS variables describing the config for that property. Looking at the `breakpoint` example again, this config  
```

Comment:
Should be generating all the 'variables' files? If so,` /source/css/sass/_variables--[propertyname].sass` or `/source/css/sass/variables/[propertyname].sass`?

(Also, `sass` inside `css`?)

Reply:
Originally I though that it'd only need to generate those that are either extended by additional config, or that contain knowledge shared between technology layers, but now I'm thinking that it should generate all of them.

Reply:
`sass` inside `css` because of how it's used during the export process. I'm not wedded to that structure, but a change is outside the scope of this PR.